packages/governance/src/binaryBallotCounter.js

```
@@ -0,0 +1,130 @@
+
+    // ballot template has position choices; Each ballot in allBallots should
+    // match. count the valid ballots and report results.
+    const [positionA, positionB] = template.getPositions();
+    let spoiled = 0n;
+    const tally = {
+      [positionA]: 0n,
+      [positionB]: 0n,
+    };
+
+    allBallots.entries().forEach(([_, { ballot, weight }]) => {
+      const choice = ballot.chosen[0];
+      if (!template.getPositions().includes(choice)) {
+        spoiled += weight;
+      } else {
+        tally[choice] += weight;
+      }
+    });
+
+    const stats = {
+      spoiled,
+      votes: allBallots.entries().length,
+      results: [
+        { position: positionA, total: tally[positionA] },
+        { position: positionB, total: tally[positionB] },
+      ],
+    };
+
+    const quorumCheck = await E(quorumChecker).check(stats);
+    if (!quorumCheck) {
+      outcomePromise.reject('No quorum');
+    }
+
+    if (tally[positionA] > tally[positionB]) {
+      outcomePromise.resolve(positionA);
+    } else if (tally[positionB] > tally[positionA]) {
+      outcomePromise.resolve(positionB);
+    } else {
+      outcomePromise.resolve(TIE_VOTE);
+    }
+
+    tallyPromise.resolve(stats);
+  };
+
+  const sharedFacet = {
+    getBallotTemplate: () => template,
+    isOpen: () => isOpen,
+    getQuestionPositions,
+  };
+
+  const voterFacet = Far('voterFacet', {
+    submitVote: recordBallot,
+  });
+
+  const creatorFacet = Far('adminFacet', {
+    ...sharedFacet,
+    closeVoting: () => {
+      isOpen = false;
+    },
+    countVotes,
+    getVoterFacet: () => voterFacet,
+  });
+
+  const publicFacet = Far('publicFacet', {
+    ...sharedFacet,
+    getOutcome: () => outcomePromise.promise,
+    getStats: () => tallyPromise.promise,
+  });
+  return { publicFacet, creatorFacet };
+};
+
+const start = zcf => {
+  const { question, positions } = zcf.getTerms();
+  return makeBinaryBallotCounter(question, positions[0], positions[1]);
```

What if positions.length > 2? Can we fail fast with an assert here?

---

I put the check as the first thing inside `makeBinaryBallotCounter()`, so it'll fail fast here, too.